src/Paystack.php

```diff
@@ -278,4 +278,264 @@ private function getData()
+            "send_sms" => request()->send_sms,
+            "currency" => request()->currency,
+        ];
+
+        $this->setRequestOptions();
+
+        return $this->setGetResponse('/plan/' . $plan_code, $data)->getResponse();
+
+    }
+
+    /**
+     * Create a customer
+     * @return array
+     */
+    public function createCustomer(){
+
+        $data = [
+            "email" => request()->email,
+            "first_name" => request()->fname,
+            "last_name" => request()->lname,
+            "phone" => request()->phone,
+            "metadata" => request()->addtional_info /* key => value pairs array */
+
+        ];
+
+        $this->setRequestOptions();
+
+        $this->response = $this->setGetResponse('/customer', $data);
+
+    }
+
+    /**
+     * Fetch a customer based on id or code
+     * @param $customer_id
+     * @return array
+     */
+    public function fetchCustomer($customer_id){
+
+        $this->setRequestOptions();
+
+        return $this->setGetResponse('/customer/'. $customer_id, [])->getResponse();
+
+    }
+
+    /**
+     * Update a customer's details based on their id or code
+     * @param $customer_id
+     * @return array
+     */
+    public function updateCustomer($customer_id){
+
+        $data = [
+            "email" => request()->email,
+            "first_name" => request()->fname,
+            "last_name" => request()->lname,
+            "phone" => request()->phone,
+            "metadata" => request()->addtional_info /* key => value pairs array */
+
+        ];
+
+        $this->setRequestOptions();
+
+        return $this->setGetResponse('/customer/'. $customer_id, $data)->getResponse();
+
+    }
+
+    /**
+     * Export tranactions in .CSV
+     * @return array
+     */
+    public function exportTransactions(){
+
+        $data = [
+            "from" => request()->from,
+            "to" => request()->to,
+            'settled' => request()->settled
+        ];
+
+        $this->setRequestOptions();
+
+        return $this->setGetResponse('/transaction/export', $data)->getResponse();
+
+    }
+
+    /**
+     * Create a subscription to a plan from a customer.
+     * @return array
+     */
+    public function createSubscription(){
+
+        $data = [
+            "customer" => request()->customer, //Customer email or code
+            "plan" => request()->plan,
+            "authorization" => request()->authorization_code
+        ];
+
+        $this->setRequestOptions();
+
+        $this->response = $this->setGetResponse('/subscription', $data);
+    }
+
+    /**
+     * Enable a subscription using the subscription code and token
+     * @return array
+     */
+    public function enableSubscription(){
+
+        $data = [
+            "code" => request()->code,
+            "token" => request()->token,
+        ];
+
+        $this->setRequestOptions();
+
+        return $this->setGetResponse('/subscription/enable', $data)->getResponse();
```

Comment:
Oga, review again abeg
 

	 	
Funsho Okubanjo
http://devfunsho.com
 <http://devfunsho.com/> <https://about.me/iamfunsho?promo=email_sig&utm_source=email_sig&utm_medium=email_sig&utm_campaign=external_links> 


> On May 22, 2016, at 11:49 PM, Prosper Otemuyiwa <notifications@github.com> wrote:
> 
> In src/Paystack.php <https://github.com/unicodeveloper/laravel-paystack/pull/4#discussion_r64157749>:
> 
> > +    }
> > +
> > +    /**
> > +     * Enable a subscription using the subscription code and token
> > +     * @return array
> > +     */
> > +    public function enableSubscription(){
> > +
> > +        $data = [
> > +            "code" => request()->code,
> > +            "token" => request()->token,
> > +        ];
> > +
> > +        $this->setRequestOptions();
> > +
> > +        return $this->setGetResponse('/subscription/enable', $data)->getResponse();
> Shouldnt this be a post method?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/unicodeveloper/laravel-paystack/pull/4/files/76faead48c5c26d42299be04726ec73473d34a92#r64157749>